### PR TITLE
Az296 merge in ri changes

### PR DESCRIPTION
**Pull request by rzezeski**
Integrated changes that Rusty made in the RI version of merge_index.

### CODE REVIEW

File: src/merge_index.erl [hunk 1]
```diff
@@ -85,12 +101,35 @@ stream(ServerPid, Index, Field, Term, Pid, Ref, FilterFun) ->
 %%
 %% `Size' - The size of the term in bytes.
 %%
-%% @see stream/7.
+%% TODO: Pull size out before pull request
```
Comment (kellymclaughlin) [inline src/merge_index.erl]: I am not sure what this TODO means and if it's intended for this pull request or another so just wanted to bring it to your attention. There's a similar one on line 124.

Comment (rzezeski) [inline src/merge_index.erl]: Yea this was when I thought we were going to pull the `Size` parameter out of the API but I decided against it.

File: test/merge_index_tests.erl [hunk 2]
```diff
@@ -96,17 +99,44 @@ postcondition(S, {call,_,is_empty,_}, V) ->
...
+    %% Assert that the weight is _greater than or equal_ b/c the
+    %% bloom filter could cause false positives.
+    ok == ?assert(W >= length(L));
 postcondition(#state{postings=Postings}, {call,_,fold,_}, {ok, V}) ->
     %% NOTE: The order in which fold returns postings is not
-    %% deterministic thus both must be sorted.
-    ok == ?assertEqual(lists:sort(Postings), lists:sort(V));
+    %% deterministic.
+
+    %% Each member of V should be a memeber of Postings, they aren't
```
Comment (kellymclaughlin) [inline test/merge_index_tests.erl]: One more _memeber_ instance.

**Comment by kellymclaughlin** [comment]
Code looks good and all tests and dialyzer pass. +1 to merge.

**Pull request merged**